frontend/src/components/dashboard/contactsHomePage/ContactHomePage.js

```diff
@@ -65,12 +57,6 @@ export class ContactHomePage extends Component {
     },
   };
 
-  componentWillMount() {
-    // Fetch dependencies from API
-    this.props.fetchPartyRelationshipTypes();
-    this.props.fetchProvinceCodes();
-  }
```

Comment:
That's what we were previously doing. The problem that started to arise was that `MineDashboard`, `Dashboard`, `CustomHomePage` etc all needed similar or different copies of the data, thus all components were fetching the staticContent directly. 

We had an isOptionsLoaded check, but that was out of sync if not all components fetched the same data, causing us to place fetches outside of the check.

This implementation could have been made better in regards to error handling/duplicates etc. 

Instead, I opted to fetch all the staticContent at the highest level parent to remove duplicate network calls, Thus removing the need for "isOptionsLoaded" since the state will always be empty when `App.js` loads proving that check to be redundant.

Reply:
That all makes sense to me, but I think the problem was the those components weren't checking the store prior to making the network requests. I don't think the problem was that multiple components required and had the ability to fetch their own external dependencies.

The "duplicate" calls should, imo, still exist, but they'd only be executed after a check of the store to see if the data needs to be fetched. This could probably even be a generalized wrapper that accepts a selector and either returns the cached value from the store or makes a network request to fetch the data if it's not in the store.

Perhaps I'll propose implementing this pattern as a hackathon project with an in-app refresh that would skip the store check and just execute the network request.

A benefit of each component having the ability to fetch its external dependencies (whether by a direct connection to Redux or via a callback that lives in a parent component) is that a standardized "refresh" method could re-fetch all of the external dependencies relevant to only that component.

Reply:
I like the idea of a general refresh re-fetch action.

I guess FetchOnMount could wrap the components themselves vs. wrapping `App.js` at which point we could utilize a check to the store before making the fetches. 🤔